File: 2022/python/day01/part1and2_test.py
import pytest
import os
from part1and2 import find_biggest_backpack, find_top_three_backpacks

@pytest.fixture
def example_input():
  return '''1000
2000
3000

4000

5000
6000

7000
8000
9000

10000
'''

@pytest.fixture
def full_input():
  dirname = os.path.dirname(__file__)
  filename = os.path.join(dirname, '../../input/day01')
  with open(filename) as file:
    full_input = file.read()
    return full_input

def test_find_most_calories(full_input, example_input):
  assert find_biggest_backpack(example_input) == 24000
  assert find_biggest_backpack(full_input) == 75622

def test_find_top_three_backpacks(full_input):
  assert find_top_three_backpacks(full_input) == 213159

File: 2022/typescript/src/day01/part1and2.test.ts
import { readFileSync } from "fs";
import { findHeaviest, findThreeHeaviest } from "./part1and2";
import path from "path";

describe("findHeaviest", () => {
  it("returns total for one elf", () => {
    expect(findHeaviest("1\n2\n3\n")).toBe(6);
  });

  it("returns bigger total out of two elves", () => {
    expect(findHeaviest("1\n2\n\n5\n6\n")).toBe(11);
  });

  it("returns bigger total out of two elves", () => {
    const filePath = path.join(__dirname, "../../../input/day01");
    const fileString = readFileSync(filePath, "utf8");
    expect(findHeaviest(fileString)).toBe(75622);
  });
});

describe("findThreeHeaviest", () => {
  it("returns biggest", () => {
    const filePath = path.join(__dirname, "../../../input/day01");
    const fileString = readFileSync(filePath, "utf8");
    expect(findThreeHeaviest(fileString)).toBe(213159);
  });
});
